pkg/interceptors/server/server.go

```diff
@@ -0,0 +1,123 @@
+package server
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/tektoncd/triggers/pkg/interceptors/bitbucket"
+	"github.com/tektoncd/triggers/pkg/interceptors/cel"
+	"github.com/tektoncd/triggers/pkg/interceptors/github"
+	"github.com/tektoncd/triggers/pkg/interceptors/gitlab"
+
+	"github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	"go.uber.org/zap"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Server struct {
+	KubeClient   kubernetes.Interface
+	Logger       *zap.SugaredLogger
+	interceptors map[string]v1alpha1.InterceptorInterface
+}
+
+func NewWithCoreInterceptors(k kubernetes.Interface, l *zap.SugaredLogger) *Server {
+	i := map[string]v1alpha1.InterceptorInterface{
+		"bitbucket": bitbucket.NewInterceptor(k, l).(v1alpha1.InterceptorInterface),
+		"cel":       cel.NewInterceptor(k, l).(v1alpha1.InterceptorInterface),
+		"github":    github.NewInterceptor(k, l).(v1alpha1.InterceptorInterface),
+		"gitlab":    gitlab.NewInterceptor(k, l).(v1alpha1.InterceptorInterface),
+	}
```

Comment:
No action needed, but something to think about: I wonder if we should have libraries Register interceptors at init so that it's easier for other core interceptors to be added opaquely. 🤔  

(this is also making me think we should look into embedding the logger into the context)

Reply:
yes, we need a way to dynamically register interceptors at startup. Once we have the interceptor config crd, we can have an informer/lister to fetch all known interceptors either on startup or on the fly. 
If we need to, we could also add a way to register onto the core interceptor deployment too using a Register func that is called at init.

Reply:
on the logging from context -- yes, I plan on plumbing it through the context once we actually wire this up in the new binary.